sound/soc/sof/debug.c

```diff
@@ -61,17 +60,18 @@ static ssize_t sof_dfsentry_read(struct file *file, char __user *buffer,
 				 "Copying cached debugfs data\n");
 			memcpy(buf, dfse->cache_buf + pos, size);
 		}
+#else
+		/* if the DSP is in D3 */
+		if (!pm_runtime_active(sdev->dev) &&
+		    dfse->access_type == SOF_DEBUGFS_ACCESS_D0_ONLY)
+			dev_dbg(sdev->dev,
+				"DSP in D3. Debugfs mem might be invalid\n");
```

Comment:
that looks like an error case to me. if we don't cache and want to read something that's not available, we should do a pm_runtime_get_sync and pm_runtime_autosuspend. Allowing to read invalid values doesn't seem quite right - or we return an error on read.
it's actually quite nice if we have 3 choices
a) caching
b) wake-up on read
c) error on read- the user needs to disable pm_runtime manually.

Reply:
@plbossart ok that makes sense. will fix in v4

Reply:
@plbossart how would the user specify option b. Would it be another Kconfig setting?

Reply:
@plbossart I think what we could do is if debugfs caching is disabled, we always wake up the DSP if the DSP is not in panic. I can check the FW status before attempting to do a pm_runtime_get_sync() to resume the DSP. 

But I do also feel that resuming the DSP just to read debugfs entries is pointless because we'd have lost all context from before anyway. Please let me know what you think